lexxpluss_apps/src/board_controller.cpp

```diff
@@ -466,32 +471,46 @@ class emergency_switch_set { // Variables Implemented
 
 class wheel_switch { // Variables Implemented
 public:
-    void set_disable(bool disable) {
-        if (disable) {
-            gpio_dt_spec gpio_dev = GET_GPIO(wheel_en);
-            if (!gpio_is_ready_dt(&gpio_dev)) {
-                LOG_ERR("gpio_is_ready_dt Failed\n");
-                return;
+    void poll() {
+        // update  
+        if (pending_left_right_disable.has_value()) {
+            auto elapsed{k_uptime_get() - last_asserted_at};
+            if (elapsed > SOFTWARE_BRAKE_DELAY_MS) {
+                left_right_disable = pending_left_right_disable.value();
+                pending_left_right_disable = std::nullopt;
             }
-            gpio_pin_set_dt(&gpio_dev, 0);
-            left_right_disable = true; 
+        }
+
+        set_disable_impl(!is_enabled());
+    }
+    void set_disable(bool disable, bool with_delay = false) {
+        if (!with_delay) {
+            left_right_disable = disable;
+            pending_left_right_disable = std::nullopt;
         } else {
-            gpio_dt_spec gpio_dev = GET_GPIO(wheel_en);
-            if (!gpio_is_ready_dt(&gpio_dev)) {
-                LOG_ERR("gpio_is_ready_dt Failed\n");
-                return;
-            }
-            gpio_pin_set_dt(&gpio_dev, 1);
-            left_right_disable = false;
+            pending_left_right_disable = disable;
+            last_asserted_at = k_uptime_get();
```

Comment:
Store specified value to `pending_left_right_disable`. And it will be used in `poll` function